modular_skyrat/modules/verbs/code/subtle.dm

```diff
@@ -83,35 +80,46 @@
 	return FALSE
 
 /datum/emote/living/subtler/run_emote(mob/user, params, type_override = null)
+	if(!can_run_emote(user))
+		to_chat(user, "You can't emote at this time.", type = MESSAGE_TYPE_WARNING)
+		return FALSE
 	var/subtler_message
 	var/subtler_emote = params
+	var/mob/target
 	if(is_banned_from(user, "emote"))
-		to_chat(user, "You cannot send subtle emotes (banned).")
+		to_chat(user, "You cannot send subtle emotes (banned).", type = MESSAGE_TYPE_WARNING)
 		return FALSE
 	else if(user.client && user.client.prefs.muted & MUTE_IC)
-		to_chat(user, "You cannot send IC messages (muted).")
+		to_chat(user, "You cannot send IC messages (muted).", type = MESSAGE_TYPE_WARNING)
 		return FALSE
-	else if(!params)
-		subtler_emote = stripped_multiline_input(user, "Choose an emote to display.", "Subtler" , null, MAX_MESSAGE_LEN)
+	else if(!subtler_emote)
+		subtler_emote = tgui_input_text(user, "Choose an emote to display.", "Subtler" , null, MAX_MESSAGE_LEN, TRUE)
 		if(subtler_emote && !check_invalid(user, subtler_emote))
-			var/type = input("Is this a visible or hearable emote?") as null|anything in list("Visible", "Hearable")
-			switch(type)
-				if("Visible")
-					emote_type = EMOTE_VISIBLE
-				if("Hearable")
-					emote_type = EMOTE_AUDIBLE
-				else
-					alert("Unable to use this emote, must be either hearable or visible.")
-					return
+			var/list/in_view = get_hearers_in_view(1, user)
+			in_view -= GLOB.dead_mob_list
+			in_view.Remove(user)
+
+			for(var/mob/inviewmob in in_view)
+				if(!istype(inviewmob))
+					in_view.Remove(inviewmob)
+			var/list/targets = list("1-Tile Range", "Same Tile") + in_view
+			target = tgui_input_list(user, "Pick a target", "Target Selection", targets)
+			switch(target)
+				if("1-Tile Range")
+					target = SUBTLE_DEFAULT_DISTANCE
+				if("Same Tile")
+					target = SUBTLE_SAME_TILE_DISTANCE
```

Comment:
Those string options should be defines, since you're using them in more than one spot, so that it's easier to modify that choice in the future, if it's ever necessary. Minor code quality things.